optee-utee/src/tee_parameter.rs

```diff
@@ -0,0 +1,302 @@
+            ParamContent::ValueInout { a, b } => Some((*a, *b)),
+            _ => None,
+        }
+    }
+
+    fn get_type(&self) -> ParamType {
+        match &self.content {
+            ParamContent::None => ParamType::None,
+            ParamContent::MemrefInput { .. } => ParamType::MemrefInput,
+            ParamContent::MemrefOutput { .. } => ParamType::MemrefOutput,
+            ParamContent::MemrefInout { .. } => ParamType::MemrefInout,
+            ParamContent::ValueInput { .. } => ParamType::ValueInput,
+            ParamContent::ValueOutput { .. } => ParamType::ValueOutput,
+            ParamContent::ValueInout { .. } => ParamType::ValueInout,
+        }
+    }
+
+    fn get_raw_type(&self) -> u32 {
+        self.get_type() as u32
+    }
+
+    fn as_raw(&mut self) -> raw::TEE_Param {
+        match &mut self.content {
+            ParamContent::None => raw::TEE_Param {
+                memref: raw::Memref {
+                    buffer: core::ptr::null_mut(),
+                    size: 0,
+                },
+            },
+            ParamContent::MemrefInput { buffer } => raw::TEE_Param {
+                memref: raw::Memref {
+                    buffer: (*buffer).as_ptr() as *mut core::ffi::c_void,
+                    size: buffer.len(),
+                },
+            },
+            ParamContent::MemrefOutput { buffer, written: _ } => raw::TEE_Param {
+                memref: raw::Memref {
+                    buffer: (*buffer).as_mut_ptr() as *mut core::ffi::c_void,
+                    size: buffer.len(),
+                },
+            },
+            ParamContent::MemrefInout { buffer, written: _ } => raw::TEE_Param {
+                memref: raw::Memref {
+                    buffer: (*buffer).as_mut_ptr() as *mut core::ffi::c_void,
+                    size: buffer.len(),
+                },
+            },
+            ParamContent::ValueInput { a, b } => raw::TEE_Param {
+                value: raw::Value { a: *a, b: *b },
+            },
+            ParamContent::ValueInout { a, b } => raw::TEE_Param {
+                value: raw::Value { a: *a, b: *b },
+            },
+            ParamContent::ValueOutput { a, b } => raw::TEE_Param {
+                value: raw::Value { a: *a, b: *b },
+            },
+        }
+    }
+
+    fn update_size_from_raw(&mut self, raw_param: &raw::TEE_Param) {
+        match &mut self.content {
+            ParamContent::MemrefOutput { buffer: _, written } => {
+                *written = unsafe { raw_param.memref.size };
+            }
+            ParamContent::MemrefInout { buffer: _, written } => {
+                *written = unsafe { raw_param.memref.size };
+            }
+            _ => {}
+        }
+    }
+
+    fn update_value_from_raw(&mut self, raw_param: &raw::TEE_Param) {
+        match &mut self.content {
+            ParamContent::ValueInout { a, b } => {
+                *a = unsafe { raw_param.value.a };
```

Comment:
I am curious why we consider accessing `raw::TEE_Param` is `unsafe` here. Can you add comment about the reason to the code?